AwardsServer/AwardsServer/SocketHandler.cs

```diff
@@ -240,20 +290,43 @@ private void NewConnections()
                         continue;
                     }
                     dataFromClient = dataFromClient.Substring(1, dataFromClient.LastIndexOf("`")-1);
+                    
                     var user = new SocketConnection(clientSocket, dataFromClient); // first message is username.
+                    lock(LockClient)
+                    {
+                        if (CurrentClients.Select(x => x.UserName).Contains(dataFromClient) || ClientQueue.Select(x => x.UserName).Contains(dataFromClient))
```

Comment:
In case its not obvious, this searches both lists to ensure the client's accountname is not already present

Depending on the options, this may randomly generate something to append and continue, or reject the connection

Reply:
amazing

Reply:
absolutely wonderful